kernel/src/hil/hasher.rs

```diff
@@ -0,0 +1,105 @@
+//! Interface for Hasher
+
+use crate::utilities::leasable_buffer::LeasableBuffer;
+use crate::ErrorCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+///
+/// 'L' is the length of the 'u8' array to store the digest output.
+pub trait Client<'a, const L: usize> {
+    /// This callback is called when the data has been added to the digest
+    /// engine.
+    /// On error or success `data` will contain a reference to the original
+    /// data supplied to `add_data()`.
+    fn add_data_done(&'a self, result: Result<(), ErrorCode>, data: &'static mut [u8]);
+
+    /// This callback is called when a digest is computed.
+    /// On error or success `digest` will contain a reference to the original
+    /// data supplied to `run()`.
+    fn hash_done(&'a self, result: Result<(), ErrorCode>, digest: &'static mut [u8; L]);
+}
+
+/// Computes a non-cryptographic hash over data
+///
+/// 'L' is the length of the 'u8' array to store the digest output.
```

Comment:
You're interchangeably using the words hash and digest here, when it seems like "digest" means cryptographic hash and "hash" means non-cryptographic hash. This looks like a copy-paste error. Please carefully read over the file and edit it.